### PR TITLE
Add sendKeys and val element methods

### PR DESCRIPTION
**Pull request by alexdavid**
@kevgo please review

### CODE REVIEW

File: features/send_keys.feature [hunk 1]
```diff
@@ -0,0 +1,11 @@
+Feature: Sending keypresses to a webpage
+
+  Background:
+    Given I create a Sanelenium instance by running "new Browser 'http://localhost:5000'"
+
+
+  Scenario: Sending keypresses to a textarea
+    Given I am on a page with the HTML "<textarea></textarea>"
+    When I run "browser.$('textarea').sendKeys 'hello browser', done"
+    Then <textarea> has the value "hello browser"
```
Comment (kevgo) [inline features/send_keys.feature]: Then the <textarea> on my page has the value "hello browser"

File: features/send_keys.feature [hunk 2]
```diff
@@ -0,0 +1,11 @@
+Feature: Sending keypresses to a webpage
+
+  Background:
+    Given I create a Sanelenium instance by running "new Browser 'http://localhost:5000'"
```
Comment (kevgo) [inline features/send_keys.feature]: What do you think about:

```cucumber
Given I create a Sanelenium instance by running "browser = new Browser 'http://localhost:5000'"

File: features/step_definitions/run_steps.coffee [hunk 3]
```diff
@@ -11,5 +11,5 @@ module.exports = ->
 
 
   @When /^I run "([^"]+)"$/, (coffeeExpression, done) ->
-    @done = sinon.spy -> done()
+    @done = sinon.spy -> setTimeout done, 10
```
Comment (kevgo) [inline features/step_definitions/run_steps.coffee]: Why do we wait 10ms here?

File: features/support/html_event_logger.coffee [hunk 4]
```diff
@@ -1,12 +1,22 @@
...
+  getEventObject = (event) ->
+    eventObject =
+        event: event.type
+        tag: event.target.tagName.toLowerCase()
+    if event.type is 'keydown'
+      eventObject.shortcut = "#{if event.ctrlKey then 'ctrl'}+#{String.fromCharCode event.keyCode}".toLowerCase()
+    eventObject
+
+
+  ['click', 'keydown'].map (eventType) ->
```
Comment (kevgo) [inline features/support/html_event_logger.coffee]: Does this have to be a `map` here? I'm unclear about the return type.

File: features/support/test_web_server.coffee [hunk 5]
```diff
@@ -46,7 +46,15 @@ class TestWebServer
     , done
 
 
-  respondWith: (@respondWithHtml) ->
+  respondWith: (respondWith) ->
```
Comment (kevgo) [inline features/support/test_web_server.coffee]: The parameter name should not match the method name. What about `respondWith: (html) ->`?

File: features/val.feature [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+  Scenario: Reading the value of <input>
+    Given I am on a page with the HTML "<input value='foo bar'>"
+    When I run "browser.$('input').val done"
+    Then the parameter "done" is called with the arguments `null, "foo bar"`
+
+
+  Scenario: Reading the value of <textarea>
+    Given I am on a page with the HTML "<textarea>fizz buzz</textarea>"
+    When I run "browser.$('textarea').val done"
+    Then the parameter "done" is called with the arguments `null, "fizz buzz"`
```
Comment (kevgo) [inline features/val.feature]: All of this is really cool!

File: src/element.coffee [hunk 7]
```diff
@@ -14,8 +26,22 @@ class Element extends Asyncify
       @element.click()
 
 
+    sendKeys: (keys) ->
+      @element.sendKeys keys
```
Comment (kevgo) [inline src/element.coffee]: Could this be simplified to 

```coffeescript
sendKeys: @element.sendKeys
```

File: src/element.coffee [hunk 8]
```diff
@@ -1,8 +1,20 @@
 Asyncify = require './asyncify'
-{By} = require 'selenium-webdriver'
+{By, Key} = require 'selenium-webdriver'
```
Comment (kevgo) [inline src/element.coffee]: `By` is unused

File: features/support/html_event_logger.coffee (hunk 4 above)
Comment (alexdavid) [inline features/support/html_event_logger.coffee]: It doesn't have to be a map, I just preferred a map over the double indentation of the alternative:
```coffeescript
for eventType in ['click', 'keydown']
  do (eventType) ->
    # ...
```

File: features/support/test_web_server.coffee (hunk 5 above)
Comment (alexdavid) [inline features/support/test_web_server.coffee]: :+1:

File: src/element.coffee (hunk 7 above)
Comment (alexdavid) [inline src/element.coffee]: No, because `this` would not point to the instance anymore

File: features/support/html_event_logger.coffee (hunk 4 above)
Comment (kevgo) [inline features/support/html_event_logger.coffee]: Could it be an `each`?

Comment (alexdavid) [inline features/support/html_event_logger.coffee]: Arrays don't have `each`. I would need to bring in underscore into the frontend which means adding an additional route for it

File: features/send_keys.feature (hunk 1 above)
Comment (alexdavid) [inline features/send_keys.feature]: Done

File: features/send_keys.feature (hunk 2 above)
Comment (alexdavid) [inline features/send_keys.feature]: Done

File: features/step_definitions/run_steps.coffee (hunk 3 above)
Comment (alexdavid) [inline features/step_definitions/run_steps.coffee]: Delay removed

**Comment by alexdavid** [comment]
@kevgo all comments addressed

File: features/support/html_event_logger.coffee (hunk 4 above)
Comment (kevgo) [inline features/support/html_event_logger.coffee]: Let's avoid underscore. What about `forEach`?

Comment (kevgo) [inline features/support/html_event_logger.coffee]: Or doing a CoffeeScript loop here?

```coffeescript
for eventType in ['click', 'keydown']
  document.addEventListener eventType, (event) ->
    ...
```

The `map` holds a semantic meaning that is inappropriate here, and it is wasteful by building extra data structures.

**Comment by kevgo** [comment]
This looks good! I think the last remaining issue is the `map` thing. Once that's resolved, this is :rocket:

File: features/support/html_event_logger.coffee (hunk 4 above)
Comment (alexdavid) [inline features/support/html_event_logger.coffee]: Ahh, forgot about `forEach`.  I'll use `forEach` instead.  Thanks!

**Pull request closed without merging**